app/models/bid.rb

```diff
@@ -11,8 +11,11 @@ class Bid < ApplicationRecord
 
   scope :this_month, -> {where(created_at: (Time.now.beginning_of_month..Time.zone.now))}
 
-  def toggle_accept(status)
-    update(accepted: !status)
+  def toggle_accept!(status)
+  	self.accepted = !status
+  	self.job.agent_id = accepted ? bidder.id : nil
+  	self.save
+  	self.job.save
```

Comment:
I just found this actually: http://api.rubyonrails.org/classes/ActiveRecord/AutosaveAssociation.html

We might be able to get away with just `self.save` :D